src/thor/triplegbuilder.cc

```diff
@@ -1344,10 +1344,10 @@ TripLeg_Edge* AddTripEdge(const AttributesController& controller,
   }
 
   if (directededge->access_restriction() && edge_itr->restriction_index != kInvalidRestriction) {
-    const std::vector<baldr::AccessRestriction>& restrictions =
-        graphtile->GetAccessRestrictions(edge.id(), costing->access_mode());
-    trip_edge->mutable_restriction()->set_type(
-        static_cast<uint32_t>(restrictions[edge_itr->restriction_index].type()));
+    auto restriction = graphtile->GetAccessRestrictionAtIndex(edge.id(), costing->access_mode(),
```

Comment:
Actually instead of `restriction_index` we could store its type only and it would be a good optimisation by itself... But I am not sure what was the plan: may be will need to have more metadata from that restriction in the future and not only type?

Reply:
true, currently that'd be fair. it's only for PathInfo which is not that critical, more a nano-optimization:)